addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -438,6 +439,48 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         }
         this._addHiddenField(value, fieldName);
     },
+    /**
+     * Prompts the user to save changes before being redirected
+     * towards an action specified in value.
+     *
+     * @see this.selectClass for parameters
+     */
+    promptSaveRedirect: function (name, value, widgetValue) {
+        return new Promise(resolve => {
+            Dialog.confirm(this, _t(
+                "Would you like to save before being redirected? Unsaved changes will be discarded."
+            ), {
```

Comment:
Not sure this will be extracted as a translatable term if written like that (in any case, why writing like that).